flocker/control/_persistence.py

```diff
@@ -160,21 +266,65 @@ def __init__(self, reactor, path):
         """
         MultiService.__init__(self)
         self._path = path
+        self._config_path = self._path.child(b"current_configuration.json")
         self._change_callbacks = []
         LeaseService(reactor, self).setServiceParent(self)
 
     def startService(self):
         if not self._path.exists():
             self._path.makedirs()
-        self._config_path = self._path.child(b"current_configuration.v1.json")
+        self.load_configuration()
+        MultiService.startService(self)
+        _LOG_STARTUP(configuration=self.get()).write(self.logger)
+
+    def _process_v1_config(self):
+        """
+        Check if a v1 configuration file exists and upgrade it if necessary.
+        After upgrade, the v1 configuration file is retained with an archived
+        file name.
+        """
+        v1_config_path = self._path.child(
+            b"current_configuration.v1.json")
+        v1_archived_path = self._path.child(
+            b"current_configuration.v1.old.json")
+        # Check for a v1 config and upgrade to latest if found.
+        if v1_config_path.exists():
+            v1_json = v1_config_path.getContent()
+            with _LOG_UPGRADE(self.logger,
+                              configuration=v1_json,
+                              source_version=1,
+                              target_version=_CONFIG_VERSION):
+                updated_json = migrate_configuration(
+                    1, _CONFIG_VERSION, v1_json,
+                    ConfigurationMigration
+                )
+                self._config_path.setContent(updated_json)
+                v1_config_path.moveTo(v1_archived_path)
```

Comment:
There doesn't seem to be much that's actually specific to v1 about this code, apart from the paths and the variable names. 

Please make these parameters and pass them in from `load_configuration`